server/src/main/java/org/elasticsearch/search/profile/query/InternalProfileCollector.java

```diff
@@ -115,7 +115,7 @@ public CollectorResult getCollectorTree() {
     }
 
     private static CollectorResult doGetCollectorTree(InternalProfileCollector collector) {
-        List<CollectorResult> childResults = new ArrayList<>(collector.children.size());
+        List<CollectorResult> childResults = new ArrayList<>(collector.children.length);
```

Comment:
I wonder if you need to check for null here now?

Reply:
forget this, just check and it get initialize, so this is fine.

Reply:
yea, I think it's a good idea.